src/charm.py

```diff
@@ -108,16 +124,28 @@ def _hydra_layer(self) -> Layer:
         }
         return Layer(layer_config)
 
+    @property
+    def _hydra_service_is_created(self) -> bool:
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            return False
+
+        return True
+
     def _render_conf_file(self) -> None:
         """Render the Hydra configuration file."""
         with open("templates/hydra.yaml.j2", "r") as file:
             template = Template(file.read())
 
         rendered = template.render(
             db_info=self._get_database_relation_info(),
-            consent_url="http://localhost:4455/consent",
-            error_url="http://localhost:4455/error",
-            login_url="http://localhost:4455/login",
+            consent_url=join(self.config.get("login_ui_url"), "consent"),
+            error_url=join(self.config.get("login_ui_url"), "oidc_error"),
+            login_url=join(self.config.get("login_ui_url"), "login"),
+            hydra_public_url=self.public_ingress.url
+            if self.model.relations["public-ingress"]
+            else f"http://127.0.0.1:{HYDRA_PUBLIC_PORT}/",
```

Comment:
I believe this should be configurable, someone might want to provide non-charmed ingress or something like that.

Most charms provide a configuration option like `external_url`, do we want to allow that?

Reply:
I'm fine with making this configurable, but in that case we'd also need to provide the same possibility for the admin endpoint and I'm not sure we want to add that many config options when we decided to recommend using charmed traefik. @massimilianogo any thoughts?

Reply:
Technically that should be true for all of the charms we have...and that is a lot of config.
In general I think that an admin will use a single ingress for the entire deployment - either our Traefik or something else.
My suggestion is to only have traefik for now and in parallel open a feature request (I can do that) in the relevant board to track the feature. If there is a need we can add it to the backlog and prioritize it accordingly.